runbot_pylint/models/runbot_build.py

```diff
@@ -0,0 +1,217 @@
+# -*- encoding: utf-8 -*-
+##############################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: moylop260@vauxoo.com
+############################################################################
+
+"""
+This module is used to create new fields in the inherited classes
+to work with pylint from runbot
+"""
+
+from openerp import models, api, fields
+import logging
+import os
+import stat
+from openerp.tools.safe_eval import safe_eval
+
+
+_logger = logging.getLogger(__name__)
+
+
+def get_depends(modules, addons_paths, depends=None):
+    """
+    Get recursive depends from addons_paths and modules list
+    """
+    if depends is None:
+        depends = []
+    for module in (modules or '').split(','):
+        if module not in depends:
+            depends.append(module)
+            for addons_path in (addons_paths or '').split(','):
+                addons_path = addons_path.strip()
+                fname_openerp = os.path.join(addons_path, module,
+                                             '__openerp__.py')
+                if os.path.isfile(fname_openerp):
+                    module_depends_list = safe_eval(
+                        open(fname_openerp, "r").read()).get('depends', [])
+                    if module_depends_list:
+                        module_depends_str = ','.join(module_depends_list)
+                        get_depends(module_depends_str, addons_paths,
+                                    depends=depends)
+    return depends
```

Comment:
This function can be simplified. It is very confusing. Here's a simplified version:
```python
from itertools import ifilter, imap

def get_depends_simplified(modules, addons_paths):
    """
    Get recursive depends from addons_paths and modules list
    :param str modules: comma separated list of modules
    :param str addons_paths: comma separated list of paths for modules
    :yields list: Unsorted list of recursive dependencies of modules
    """
    modules = set(modules.split(","))
    addons_paths = set(map(str.strip, (addons_paths or '').split(',')))
    visited = set()
    while(modules - visited):
        module = (modules - visited).pop()
        visited.add(module)
        manifest_path = os.path.join(module, '__openerp__.py')
        manifest_file = next(ifilter(
            os.path.isfile, imap(lambda p: os.path.join(p, manifest_path),
            addons_paths)
        ))
        if not manifest_file:
            # For some reason the module wasn't found
            continue
        manifest = safe_eval(open(manifest_file).read())
        modules.update(manifest.get('depends', []))
        yield module  # To make into a generator
    # return sorted(modules)  # To make a function that returns a sorted list
```